Remove unnecessary supports_not

kbrock opened the pull request:
I rebased https://github.com/ManageIQ/manageiq/pull/21478 and due to `stale` needed to open a new PR. This moves that code over here.

# Goal

Since `supports?(:unknown_feature) == false`, we no longer need to define supports up front.

The only `supports_not` I kept was in automate manager - `supports_not :refresh_ems`.

update:
- rebased #21478
- removed a few more `supports_not` entries

# Tangents / future

I think we can drop many of the `:reason` values behind `supports_not`, especially around `this feature is not supported by this provider`. These details seem unnecessary.

Fryguy commented on app/models/ext_management_system.rb:
```diff
@@ -142,37 +142,8 @@ def self.create_from_params(params, endpoints, authentications)
-  supports_not :provisioning
-  supports_not :publish
-  supports_not :reconfigure_disks
+  # NOTE: automate (a child) does not support this
```
automate or AutomationManager? (they are 2 different things, so want to be accurate, because it reads strange right now.  Separately, why wouldn't an automation manager support refresh ems?

Fryguy replied:
Oh I see, specifically embedded ansible (`ManageIQ::Providers::EmbeddedAnsible::AutomationManager`) doesn't support this.  I would remove this comment, because it feels like Basmati Rice - we may have others in the future and this comment will get stale, or perhaps we may change embedded ansible to allow "refresh".  For example, embedded ansible works on repositories, so maybe a "refresh" updates the content from the git repository.

kbrock replied:
thnx.

feels like we should not give and take away.
Feels like we should only give

But I guess the embedded ansible automation manager is a strange case, so it is good.
I don't think I made another mistake like this. But there is always that possibility.

agrare replied:
Yeah I agree I don't think we should say we support something in a base class unless it is implemented at the base class level which this is not.

agrare replied:
@kbrock you want to tackle that in a follow-up since it isn't directly related to what you're doing here?

agrare commented:
@kbrock looks good once you fix the regions spec to use `supports?(:regions)` instead of `supports_regions?`

kbrock commented:
update:
- rebased
- separate commit dropping special default unsupported reason text
- fixed vm/snapshot (removed supports that were using unsupported_reason)

agrare commented:
@miq-bot cross-repo-tests /all
